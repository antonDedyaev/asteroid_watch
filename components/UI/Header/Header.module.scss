.headerContainer {
	width: 375px;
	display: flex;
	flex-direction: column;

	align-items: flex-start;
	gap: 8px;
	padding: 16px;

	&__title {
		width: 288px;
		color: $orange;
		font-family: Passion One;
		font-size: 32px;
		font-style: normal;
		font-weight: 400;
		line-height: normal;
	}
	&__body {
		display: flex;
		flex-direction: column;
		align-self: stretch;

		color: $white-base;
		font-family: Helvetica;
		font-size: 16px;
		font-style: normal;
		font-weight: 400;
		line-height: 24px;
	}
}
