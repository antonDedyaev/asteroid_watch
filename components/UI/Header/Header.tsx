import styles from "./Header.module.scss";

const Header = () => {
	return (
		<div
			className={styles.headerContainer}
			role='header'
		>
			<div className={styles.headerContainer__title}>ARMAGEDDON 2023</div>
			<div className={styles.headerContainer__body}>
				<span>ООО “Команда им. Б. Уиллиса”.</span>
				<span>Взрываем астероиды с 1998 года.</span>
			</div>
		</div>
	);
};

export default Header;
