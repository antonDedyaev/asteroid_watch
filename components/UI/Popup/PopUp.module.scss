.popupContainer {
	height: 100vh;
	width: 100vw;
	position: fixed;
	display: flex;
	justify-content: center;
	align-items: center;
	top: 0;
	left: 0;

	background: rgba(0, 0, 0, 0.8);
	opacity: 0;
	pointer-events: none;
	transition: 0.4s;
	z-index: 5;

	&_active {
		opacity: 1;
		pointer-events: all;
	}

	&__content {
		padding: 30px 20px;
		border-radius: 10px;
		background: $dark;

		transform: scale(0.5);
		transition: all 0.4s;

		font-family: Passion One;
		font-weight: 700;
		font-size: 24px;
		letter-spacing: 1.5px;
		color: $orange;

		&_active {
			transform: scale(1);
			transition: all 0.4s;
		}
	}
}

// @media screen and (max-width: 620px) {
// 	.popupWrapper {
// 		&__content {
// 			padding: 0;
// 		}
// 	}
// }
// @media screen and (max-width: 420px) {
// 	.popupWrapper {
// 		height: 100%;
// 		width: 100%;
// 	}
// }
