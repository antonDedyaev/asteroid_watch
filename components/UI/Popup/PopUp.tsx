import styles from "./PopUp.module.scss";
import { ReactNode } from "react";

interface IPopupProps {
	isActive: boolean;
	setIsActive: (state: boolean) => void;
	children: ReactNode;
}

const Popup = ({ isActive, setIsActive, children }: IPopupProps) => {
	return (
		<div
			className={
				isActive
					? [styles.popupContainer, styles.popupContainer_active].join(" ")
					: styles.popupContainer
			}
			onClick={() => setIsActive(false)}
		>
			<div
				className={
					isActive
						? [
								styles.popupContainer__content,
								styles.popupContainer__content_active,
						  ].join(" ")
						: styles.popupContainer__content
				}
				onClick={(e) => e.stopPropagation()}
			>
				{children}
			</div>
		</div>
	);
};
export default Popup;
