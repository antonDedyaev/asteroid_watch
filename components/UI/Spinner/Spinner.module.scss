.spinnerContainer::after {
	content: "";
	display: flex;
	margin: 50px 0;
	width: 50px;
	height: 50px;
	border: 5px solid $dark;
	border-top-color: $orange;
	border-bottom-color: $orange;
	border-radius: 50%;
	-webkit-animation: loading;
	animation: loading 1s linear infinite;
}

@-webkit-keyframes loading {
	to {
		transform: rotate(1turn);
	}
}

@keyframes loading {
	to {
		transform: rotate(1turn);
	}
}

@media screen and (max-width: 930px) {
	.spinnerContainer::after {
		margin-bottom: 100px;
	}
}
