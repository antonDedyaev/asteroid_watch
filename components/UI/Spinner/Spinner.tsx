import styles from "./Spinner.module.scss";

const Spinner = () => {
	return <div className={styles.spinnerContainer}></div>;
};

export default Spinner;
