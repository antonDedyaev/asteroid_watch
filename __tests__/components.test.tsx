import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "@/components/UI/Header/Header";
import Popup from "@/components/UI/Popup/PopUp";
import Spinner from "@/components/UI/Spinner/Spinner";

describe("UI COMPONENTS SHAPSHOT TESTS", () => {
	it("renders Header unchanged", () => {
		const { container } = render(<Header />);
		expect(container).toMatchSnapshot();
	});

	it("renders PopUp unchanged", () => {
		render(
			<Popup
				isActive={true}
				setIsActive={(): void => console.log("Test")}
			>
				Предупреждающая модалка
			</Popup>
		);
		const modal = screen.getByText("Предупреждающая модалка");
		expect(modal).toMatchSnapshot();
	});

	it("renders Spinner unchanged", () => {
		const { container } = render(<Spinner />);
		expect(container).toMatchSnapshot();
	});
});
