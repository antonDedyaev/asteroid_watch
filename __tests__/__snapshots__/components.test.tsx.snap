// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI COMPONENTS SHAPSHOT TESTS renders Header unchanged 1`] = `
<div>
  <div
    class="headerContainer"
    role="header"
  >
    <div
      class="headerContainer__title"
    >
      ARMAGEDDON 2023
    </div>
    <div
      class="headerContainer__body"
    >
      <span>
        ООО “Команда им. Б. Уиллиса”.
      </span>
      <span>
        Взрываем астероиды с 1998 года.
      </span>
    </div>
  </div>
</div>
`;

exports[`UI COMPONENTS SHAPSHOT TESTS renders PopUp unchanged 1`] = `
<div
  class="popupContainer__content popupContainer__content_active"
>
  Предупреждающая модалка
</div>
`;

exports[`UI COMPONENTS SHAPSHOT TESTS renders Spinner unchanged 1`] = `
<div>
  <div
    class="spinnerContainer"
  />
</div>
`;
